test_agent/rust/src/utils.rs

```diff
@@ -0,0 +1,552 @@
+            .and_then(|entity| entity.get("version_minor"))
+        {
+            Some(_entity_version_minor) => _entity_version_minor
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _entity_special_fields = SpecialFields::default();
+        let mut _entity = UEntity::new();
+        _entity.name = _entity_name.unwrap_or_default().to_string();
+        if!(_entity_id == 0)
+        {
+        _entity.id = Some(_entity_id)
+        };
+        if!(_entity_version_major == 0)
+        {
+        _entity.version_major = Some(_entity_version_major)
+        };
+        if!(_entity_version_minor == 0)
+        {
+        _entity.version_minor = Some(_entity_version_minor)
+        };
+        _entity.special_fields = _entity_special_fields;
+      //  dbg!("_entity: {:?}", _entity);
+        let ___entity = MessageField(Some(Box::new(_entity)));
+
+        let _resource_name = match value
+            .get("resource")
+            .and_then(|resource| resource.get("name"))
+        {
+            Some(_resource_name) => _resource_name.as_str().expect("issue in name"),
+            None => "default",
+        };
+        let _resource_instance = match value
+            .get("resource")
+            .and_then(|resource| resource.get("instance"))
+        {
+            Some(_resource_instance) => _resource_instance.as_str().map(|s| s.to_owned()),
+            None => Some(String::from("default")),
+        };
+        let _resource_message = match value
+            .get("resource")
+            .and_then(|resource| resource.get("message"))
+        {
+            Some(_resource_message) => _resource_message.as_str().map(|s| s.to_owned()),
+            None => Some(String::from("default")),
+        };
+        let _resource_id = match value
+            .get("resource")
+            .and_then(|resource| resource.get("id"))
+        {
+            Some(_resource_id) => _resource_id
+                .clone()
+                .as_str()
+                .expect("not a string")
+                .parse::<u32>()
+                .expect("issue in converting to u32"),
+            None => 0,
+        };
+        let _resource_special_fields = SpecialFields::default();
+        let mut _resource = UResource::new();
+        _resource.name = _resource_name.to_owned();
+        _resource.instance = _resource_instance;
+        _resource.message = _resource_message;
+        if!(_resource_id == 0)
+        {
+        _resource.id = Some(_resource_id)
+    };
+
+        let ___resource = MessageField(Some(Box::new(_resource)));
+        let _special_fields = SpecialFields::default();
+    let mut _uuri:UUri = UUri::new();
+    if!(_authority_name.clone() == Some("default") && _authority_number_id.clone() == vec![0] && _authority_number_ip.clone() == vec![0]){
```

Comment:
For example if you take the suggestions from above, this could be:
```suggestion
    if authority.name.has_some() && authority.number.has_some() {
```